### ISSUE TITLE
It won't run on K8s 1.16 or later

### ISSUE THREAD

**Issue by monaka** [teamhephy/router#51]
Because of API deprecation since 1.16, the router can't build model.

My router's log on K8s 1.16.10 is like this.

```
$ kubectl logs -n deis              deis-router-666dfdcf55-prptk
2020/08/01 05:50:44 INFO: Starting nginx...
2020/08/01 05:50:44 INFO: nginx started.
2020/08/01 05:50:44 Error building model; not modifying certs or configuration: the server could not find the requested resource.
2020/08/01 05:50:54 Error building model; not modifying certs or configuration: the server could not find the requested resource.
2020/08/01 05:51:04 Error building model; not modifying certs or configuration: the server could not find the requested resource.
2020/08/01 05:51:14 Error building model; not modifying certs or configuration: the server could not find the requested resource.
```

```
$ kubectl get po -n deis              deis-router-666dfdcf55-prptk
NAME                           READY   STATUS    RESTARTS   AGE
deis-router-666dfdcf55-prptk   0/1     Running   9          17m
```

**Comment by Cryptophobia** [comment]
Yes, we need to update to use the new golang kubernetes-client version and update our code in the teamhephy/router related to the api deprecations here: https://github.com/teamhephy/controller/pull/110

In the meantime you could run Kubernetes with the api deprecations disabled.

**Comment by kingdonb** [comment]
I have a WIP release-type chart thing open currently at https://github.com/kingdonb/gitops-hephy/tree/master/workflow

I am working out of there, to demonstrate the issues that remain currently open before we can release v2.22. This is one of them. If you clone and take a copy of the workflow/ dir there, you can install it and help us narrow down the remaining issues! Router is one where I have to admit I have less understanding myself, I saw this issue too, before moving on to verify the experimental native ingress support was a serviceable replacement.

There are still issues in the controller, router, and likely monitor and others too. I've been wrapped up in a fair amount of personal and work stuff lately, so this hasn't got the right kind of attention yet, but we are looking forward to fixing these few remaining issues very soon! PRs are much appreciated.

If you can demonstrate a fix for this it will make it easier for us to review and incorporate it quickly into the next release. A PR against router and gitops-hephy (or both) will give me the most mileage, but any part of that which gets us closer would be much appreciated! Thanks for the report.

**Comment by dmcnaught** [comment]
I see this with kops 1.18.0 and K8s 1.18.8, and I do have the runtimeconfig set to allow the deprecations:
```
    runtimeConfig:
        extensions/v1beta1/daemonsets: "true"
        extensions/v1beta1/deployments: "true"
        extensions/v1beta1/networkpoliciesr: "true"
        extensions/v1beta1/replicasets: "true"
```

**Comment by dmcnaught** [comment]
It was working fine with 1.16 and 1.17

**Comment by kingdonb** [comment]
This has to be resolved as some cloud providers have probably already dropped 1.15, users of those clouds might not be able to enable deprecated APIs. Until this all gets handled, we may be really limiting where we can put new Workflow installations.

**Comment by Cryptophobia** [comment]
This is fixed in new Hephy Workflow release v2.22.0 going forward :)

Fixed in this PR https://github.com/teamhephy/router/pull/52
